### ISSUE TITLE
Invalid expressions in permission field presets causes 403 error in GraphQL endpoint for non-admin users

### ISSUE THREAD

**Issue by takerusilt** [directus/directus#23745]
### Describe the Bug

Accessing /graphql endpoint with non-admin permission resulted in following error regardless of the query
**when invalid expressions appears in permission entries** where **invalid** means referencing an inexist field e.g. `name` in `directus_users` which is referenced by `$CURRENT_USER`

![image](https://github.com/user-attachments/assets/21d9f8f1-62b8-4739-a885-8208debb01d2)

```text
DEBUG: You don't have permission to access field "name" in collection "directus_users" or it does not exist. Queried in root.
err: {
  "type": "",
  "message": "You don't have permission to access field \"name\" in collection \"directus_users\" or it does not exist. Queried in root.",
  "stack":
      DirectusError: You don't have permission to access field "name" in collection "directus_users" or it does not exist. Queried in root.
          at createFieldsForbiddenError (...)
          at validatePathExistence (...)
          at processAst (...)
          at UsersService.readByQuery (...)
          at process.processTicksAndRejections (...)
          at async UsersService.readOne (...)
          at async (...)
          at async fetchContextData (...)
          at async fetchDynamicVariableContext (...)
          at async fetchPermissions (...)
  "name": "DirectusError",
  "extensions": {
    "reason": "You don't have permission to access field \"name\" in collection \"directus_users\" or it does not exist. Queried in root."
  },
  "code": "FORBIDDEN",
  "status": 403
}
```

Prior to 11.0, the invalid expression did not cause error in /graphql or other endpoints

### To Reproduce

1) Create a policy, for any one collection, add the following field preset

```json
{
    "user_name": "$CURRENT_USER.name"
}
```

2) Assign policy to a non-admin user
3) Use the non-admin user's credential to issue a GraphQL query to the `/graphql` endpoint with arbitrary query

### Directus Version

v11.1.0

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by hanneskuettner** [comment]
What is the expected behavior in this case? Having it silently fail in the background doesn't seem like the best option, as that would never alert you to the underlying problem that exists and might result in to permissive permissions, allowing unwanted access. 

I agree that the error message could be improved a lot, letting you know from which policy / permission the problem originates.

**Comment by takerusilt** [comment]
In my opinion, field preset is used for initializing an item. It does not contribute itself as part of the computed permission. For scenario other than creating item, field preset should be irrelevant.

When `fetchPermissions` calling `fetchDynamicVariableContext`, it would be more logical to ignore dynamic variables living in field preset, as the field preset is not interpolated anyway, 1) it does not affect the result, 2) nor any restricted data would leak to consumer of  `fetchPermissions`.

Nonetheless if error should be raised, raising it during item creation or interpolation would be sufficient. In such cases caller of `fetchDynamicVariableContext` will include dynamic variables in the field preset as concerned.

**Comment by hanneskuettner** [comment]
> as the field preset is not interpolated anyway

That is not correct, you can use dynamic variables in field presets and they are correctly interpolated. 

The problem is that a permission in core always consists of the permission filter and the presets. So when loading them, like is required for the schema generation of the GraphQL schema they are always loaded in full.

I can see your objection so, that it should realistically only error on item creation.

**Comment by takerusilt** [comment]
Yes that would be the realistic handling. Thank you very much.

> That is not correct, you can use dynamic variables in field presets and they are correctly interpolated.

Sorry I was overlooked during debugging. I saw dynamic variables in presets are also processed in code.

**Comment by Nitwel** [comment]
So after checking everything myself, I can't seem to see a way where we could improve anything here. The "get/query" requests go through just fine and skip the access of presets. The only things that are not working is `/permissions/me` and the GQL schema generation which makes sense as they're not exclusive to only read permissions.

In that sense, I'll close this but I'm happy to reopen if there is something to improve.
